src/server/service/user.ts

```diff
@@ -1,19 +1,18 @@
 import bcrypt from 'bcrypt';
 
-import UserModel from '../models/user';
+import UserModel, { Role } from '../models/user';
 
 import type { User } from '../models/user';
 import type { ILoggerService } from './logger';
 
 export type CreateUserDTO = {
-  firstName: string;
-  lastName: string;
   email: string;
   password: string;
+  role: Role;
 };
 
 export interface IUserService {
-  create(dto: CreateUserDTO): Promise<User>;
+  create(dto: CreateUserDTO, email: string): Promise<User>;
```

Comment:
This would change to be an instance of `User` here, and the order also changes as we require the "current user" to proceed with the second argument which is "CreateUserDto".